library(numbat)
library(dplyr)
library(data.table)
library(Seurat)
'%notin%' <- Negate('%in%')



allele_counts <- as.data.frame(fread("preprocessing/SarcomaS956GEX/SarcomaS956Out/SarcomaS956_allele_counts.tsv"))


    
immune<-c('CD4+ T-cells','CD4+ Tcm','CD4+ Tem','CD8+ T-cells','CD8+ Tcm','CD8+ Tem','Class-switched memory B-cells','DC','Eosinophils','Macrophages','Macrophages M1','Macrophages M2','Memory B-cells','Monocytes','naive B-cells','Neutrophils','NK cells','Plasma cells','Tregs')



sarc <- readRDS("data_SarcomaS956GEX_genes_300_UMI_600_annotated_for_infercnv.rds")




notum<-subset(sarc,subset = celltype_ %in% immune)

cell_annot<-data.frame(cell=rownames(notum@meta.data),group=notum@meta.data$celltype_)
ref_mat<-notum@assays$RNA@counts
ref_internal = aggregate_counts(ref_mat, cell_annot)



df_sparse<-sarc@assays$RNA@counts

allele_counts <- subset(allele_counts, cell %in% rownames(sarc@meta.data))


out = run_numbat(count_mat=df_sparse, # gene x cell integer UMI count matrix 
                 lambdas_ref=ref_internal, # reference expression profile, a gene x cell type normalized expression level matrix
                 df_allele=allele_counts, # allele dataframe generated by pileup_and_phase script
                 out_dir= paste0('refNumOutput/S956'),
                 genome = "hg38",
                 ncores = 4,
                 plot = TRUE)

saveRDS(out,paste0('refNumOutput/S957_numbat_out.rds'))















